internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -244,10 +244,15 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
 		r.recorder.Eventf(controllerRollout, corev1.EventTypeNormal, "Deleting", "Deleting NumaflowControllerRollout")
 		if controllerutil.ContainsFinalizer(controllerRollout, finalizerName) {
 			ppnd.GetPauseModule().DeletePauseRequest(controllerKey)
+			// Check if dependent resources are deleted, if not then requeue after 5 seconds
+			if !r.isDependentResourceDeleted(ctx, controllerRollout) {
```

Comment:
I like your idea to be dependent only on the pipelines/isbsvc/monovertices which are children of rollout objects, in the off chance that there were another resource that were not owned by a Numaplane rollout. (Although, there are other places in the code in which we do just assume that any we find are owned by Numaplane.)

Reply:
nit: can we rename the function to `areDependentResourcesDeleted()`?